### ISSUE TITLE
Gradient Penalty with Auxiliary classifier

### ISSUE THREAD

**Issue by dougsouza** [LMescheder/GAN_stability#10]
Gradient penalty claculation should change when D has an auxiliary classifier or the gradient penalty should only be calculated over the real/fake output? Is there any intuition behind of whatever is the best solution?

Thanks!

**Comment by Johnson-yue** [comment]
@dougsouza  do you implement it？？

**Comment by dougsouza** [comment]
@Johnson-yue, yes. I tried gradient penalty only in the real/fake output of the network. But I didn't pursue much, I went back to spectral normalization.

**Comment by Johnson-yue** [comment]
Maybe spectral normalization is in common use
